File: package/node/src/modules/device.ts
import type { Transport } from "../core/transport";
import type { SessionManager } from "../core/session";
import type { DeviceInfo, DeviceStatusResponse } from "../types";
import { SessionExpiredError } from "../errors";

interface DeviceModuleDeps {
  transport: Transport;
  session: SessionManager;
}

class DeviceModule {
  private transport: Transport;
  private session: SessionManager;

  constructor(deps: DeviceModuleDeps) {
    this.transport = deps.transport;
    this.session = deps.session;
  }

  async detect(): Promise<DeviceInfo[]> {
    const accessToken = this.session.getAccessToken();

    if (!accessToken) {
      throw new SessionExpiredError();
    }

    return this.transport.get<DeviceInfo[]>("/api/v1/devices", accessToken);
  }

  async status(): Promise<DeviceStatusResponse> {
    const accessToken = this.session.getAccessToken();

    if (!accessToken) {
      throw new SessionExpiredError();
    }

    return this.transport.get<DeviceStatusResponse>(
      "/api/v1/devices/status",
      accessToken,
    );
  }
}

export { DeviceModule };

File: package/python/aether_identity/modules/device.py
"""Aether Identity Python SDK - Device module."""

from typing import List
from ..core.transport import Transport
from ..core.session import SessionManager
from ..types import DeviceInfo, DeviceStatusResponse


class DeviceModule:
    """Device management module."""

    def __init__(self, transport: Transport, session_manager: SessionManager):
        """Initialize device module.

        Args:
            transport: HTTP transport instance
            session_manager: Session manager instance
        """
        self._transport = transport
        self._session = session_manager

    def detect(self) -> List[DeviceInfo]:
        """Get list of user devices.

        Returns:
            List of device information
        """
        response = self._transport.get(
            "/api/v1/devices", access_token=self._session.get_access_token()
        )

        if isinstance(response, list):
            return [DeviceInfo.model_validate(device) for device in response]

        return []

    def status(self) -> DeviceStatusResponse:
        """Get device availability status.

        Returns:
            Device status response
        """
        response = self._transport.get(
            "/api/v1/devices/status", access_token=self._session.get_access_token()
        )

        return DeviceStatusResponse.model_validate(response)
